project/leeshengcian/README.md

```diff
@@ -0,0 +1,74 @@
+# Visualization-of-Dijkstra-Algorithm
+This Proposal shows the work flow and plans for the 2023 Autumn NYCU NSD term project
+
+## Basic Information
+
+GitHub Link : [https://github.com/leeshengcian/Visualization-of-Dijkstra-Algorithm/tree/main]
+
+Dijkstra's algorithm is an algorithm for finding the shortest paths between 
+nodes in a weighted graph, which may represent, for example, road networks. 
+It was conceived by computer scientist Edsger W. Dijkstra in 1956 and published three years later.
+
+## Problem to Solve
+
+For a given source node in the graph, the algorithm finds the shortest path between 
+that node and every other. It can also be used for finding the shortest paths from 
+a single node to a single destination node by stopping the algorithm 
+once the shortest path to the destination node has been determined.
+
+possible usage:
+
+1. Finding the shortest path of two intersections on a city map
```

Comment:
Could you please speculate how the code may be extended for the route-planning?

Reply:
I will attend this week meetup to discuss the further extension, thank you.

Reply:
While we can meet and talk, why don't you try to write down something before that?

Reply:
Through Visualization, I can draw a minimum spanning tree of a given graph and imply the path from source node to objective node(suppose objective node is node_#1, the suggested path is: source -> node_#2 -> node_#4 -> node_#1)and also show the visualization of the path

Reply:
How large a graph you can draw in a GUI interactively?

Reply:
Consider large amount of nodes, visualization has its limitation
However in such case, even plain text could seem lengthy and annoying
The image below is closer to my imagination about this system
In this map, it must contains a large number of nodes, but as we zoom out the map for wider perspective, we can somehow downscale the map, where we don't need to consider detailed nodes in path finding problem
![image](https://github.com/yungyuc/nsdhw_23au/assets/145427848/8341a3d3-4384-4a4b-b40e-62f5c88635a0)
My point is, visualization is the most simple way for user to understand the status, in that case, I wouldn't directly show, say 10000 nodes, to the user

Reply:
You are right that visualization needs special care of itself.  Many games needs that kind of level of detail control.  Based on the observation you may see a lot of possible projects for coding.

It should be fun.  Even though you do not pursue the things in this course, you may keep it in mind.